src/git.js

```diff
@@ -15,52 +15,67 @@
  */
 
 const {Owner, createOwnersMap} = require('./owner');
-const bb = require('bluebird');
 const child_process = require('child_process');
 const yaml = require('yamljs');
 const path = require('path');
-const exec = bb.promisify(child_process.exec);
-const fs = bb.promisifyAll(require('fs'));
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+const fs = require('fs').promises;
 
 const matcher = /your branch is up-to-date|your branch is up to date/i;
 
 function yamlReader(str) {
   return yaml.parse(str);
 }
 
-/**
- * Reads the actual OWNER file on the file system and parses it using the
- * passed in `formatReader` and returns an `OwnersMap`.
- */
-function ownersParser(formatReader, pathToRepoDir, ownersPaths) {
-  const promises = ownersPaths.map(ownerPath => {
-    const fullPath = path.resolve(pathToRepoDir, ownerPath);
-    return fs.readFileAsync(fullPath).then(file => {
-      return new Owner(formatReader(file.toString()), pathToRepoDir, ownerPath);
-    });
-  });
-  return bb.all(promises).then(createOwnersMap);
-}
+class Git {
 
-export class Git {
+  constructor(context) {
+    this.context = context;
+  }
+
+  /**
+   * Reads the actual OWNER file on the file system and parses it using the
+   * passed in `formatReader` and returns an `OwnersMap`.
+   */
+  async ownersParser(formatReader, pathToRepoDir, ownersPaths) {
+    const promises = ownersPaths.map(ownerPath => {
+      const fullPath = path.resolve(pathToRepoDir, ownerPath);
+      return fs.readFile(fullPath).then(file => {
+        const config = formatReader(file.toString());
+        if (!config) {
+          const str = `No config found for ${fullPath}`;
+          this.context.log.error(str);
+          // This handles OWNERS.yaml files that are empty.
```

Comment:
Future task: Print out a warning message that shows up in the checks tab if say, an entire extension does not have an `OWNERS.yaml` file.

Reply:
thats a good feature 👍

Reply:
created issue here https://github.com/google/github-owners-bot/issues/49